File: Árbol Binario Interactivo/Arbol.py
import sys

class Node:
    def __init__(self, data):
        self.data = data
        self.left = None
        self.right = None

def create_tree():
    """Función para crear un árbol binario de forma recursiva (interactiva)"""
    while True:
        try:
            print("Presiona 0 para salir")
            print("Presiona 1 para un nuevo nodo")
            choice = int(input("Ingresa la opción: "))
            
            if choice == 0:
                return None
            elif choice == 1:
                data = int(input("Ingresa los datos: "))
                temp = Node(data)
                
                print(f"Ingresa el nodo hijo izquierdo de {data}")
                temp.left = create_tree()
                
                print(f"Ingresa el nodo hijo derecho de {data}")
                temp.right = create_tree()
                
                return temp
            else:
                print("Opción no válida. Intenta de nuevo.")
        except ValueError:
            print("Entrada no válida. Por favor, ingresa 0 o 1.")
            # Continuará el bucle para pedir una nueva entrada

# Ejemplo de uso:
# root = create_tree()

File: Árbol Binario Interactivo/Arbol.js
// Para usar la entrada interactiva en JavaScript, generalmente se usa Node.js con 'readline' o 'prompt'
const readline = require('readline');
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

class Node {
    constructor(data) {
        this.data = data;
        this.left = null;
        this.right = null;
    }
}

// Función recursiva con callbacks para manejar la entrada asíncrona de Node.js
function createTreeAsync(callback) {
    console.log("Presiona 0 para salir");
    console.log("Presiona 1 para un nuevo nodo");
    
    rl.question("Ingresa la opción: ", (choiceInput) => {
        const choice = parseInt(choiceInput);

        if (choice === 0) {
            return callback(null);
        } else if (choice === 1) {
            rl.question("Ingresa los datos: ", (dataInput) => {
                const data = parseInt(dataInput);
                const temp = new Node(data);

                console.log(`Ingresa el nodo hijo izquierdo de ${data}`);
                createTreeAsync((leftNode) => {
                    temp.left = leftNode;

                    console.log(`Ingresa el nodo hijo derecho de ${data}`);
                    createTreeAsync((rightNode) => {
                        temp.right = rightNode;
                        callback(temp);
                    });
                });
            });
        } else {
            console.log("Opción no válida. Intenta de nuevo.");
            createTreeAsync(callback);
        }
    });
}

/*
// Ejemplo de uso:
// createTreeAsync((root) => {
//     console.log("Árbol creado:", root);
//     rl.close();
// });
*/